From 89a46905b5a35e8f9d4598b7220477c123e61b1d Mon Sep 17 00:00:00 2001
From: Meron Cohen <mecohen@habana.ai>
Date: Tue, 15 Apr 2025 00:42:40 +0300
Subject: [PATCH] [SW-225930]: g3: discard sec-psu from aip util

Since the compute engines draw the necessary power from the main PSU
only, there is to update the AIP utilization function by discarding the
secondary PSU power and max power from it.

Change-Id: Ie3b57546e5b158fb31ae2b0d62bede18a7a399c9
Signed-off-by: Meron Cohen <mecohen@habana.ai>
---

diff --git a/src/hlml.c b/src/hlml.c
index 06dd93e..99c3db6 100644
--- a/src/hlml.c
+++ b/src/hlml.c
@@ -431,7 +431,7 @@
 	hlml_return_int_t ret_int = HLML_INT_SUCCESS;
 	hlml_return_t ret = HLML_SUCCESS;
 	bool idle;
-	unsigned int max_power_mw, max_idle_power_mw;
+	unsigned int max_idle_power_mw;
 	int dividend, divisor;
 
 	INIT_CHECK
@@ -466,24 +466,23 @@
 
 	if (aip_info->use_new_utilization_formula) {
 		/*
-		* The utilization formula is:
-		* utilization [%] = ROUND_TO_NEAREST_FIVE(100 * dividend / divisor)
-		*
-		* Where:
-		*
-		* 1. dividend [mW] = power-consumption from both PSUs - power-consumption in IDLE state.
-		*    (In case the dividend is non-positive, then the utilization is set to 0%)
-		*
-		* 2. divisor [mW] = max-power of both PSUs - power-consumption in IDLE state.
-		*    (The divisor cannot be 0, so the division operation is safe)
-		*/
-		max_power_mw = max_power + WATT_TO_MW(aip_info->psu_sec_info->psu_sec_max_power);
+		 * The utilization formula is:
+		 * utilization [%] = ROUND_TO_NEAREST_FIVE(100 * dividend / divisor)
+		 *
+		 * Where:
+		 *
+		 * 1. dividend [mW] = power-consumption from main PSU - power-consumption in IDLE state.
+		 *    (In case the dividend is non-positive, then the utilization is set to 0%)
+		 *
+		 * 2. divisor [mW] = max-power of main PSU - power-consumption in IDLE state.
+		 *    (The divisor cannot be 0, so the division operation is safe)
+		 */
 		max_idle_power_mw = WATT_TO_MW(aip_info->max_idle_power);
-		dividend = WATT_TO_MW(aip_info->total_power) - max_idle_power_mw;
-		divisor = max_power_mw - max_idle_power_mw;
+		dividend = WATT_TO_MW(aip_info->psu_main_info.psu_main_power) - max_idle_power_mw;
+		divisor = max_power - max_idle_power_mw;
 		utilization->aip = dividend > 0 ? ROUND_TO_NEAREST_FIVE(100 * dividend / divisor) : 0;
-		HLML_LOG_DBG("total-power-consumption [W] = %u, total-max-power [W] = %u",
-			aip_info->total_power, (unsigned int)MW_TO_WATT(max_power_mw));
+		HLML_LOG_DBG("main-power-drawn [W] = %u, main-max-power [W] = %u",
+			aip_info->psu_main_info.psu_main_power, (unsigned int)MW_TO_WATT(max_power));
 		HLML_LOG_DBG("max-idle-power [W] = %u, aip-utilization = %u",
 			aip_info->max_idle_power, utilization->aip);
 	} else {
